tests/unit/lms/db/_locks_test.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+
+from lms.db import LockType, TryLockError, try_advisory_transaction_lock
+
+
+class TestTryAdvisoryTransactionLock:
+    def test_it(self, db_session, other_db_session):
+        # Initial lock attempt should succeed.
+        try_advisory_transaction_lock(db_session, LockType.OAUTH2_TOKEN_REFRESH, 123)
+
+        # Another session attempting a conflicting lock should fail
+        with pytest.raises(TryLockError):
+            try_advisory_transaction_lock(
+                other_db_session, LockType.OAUTH2_TOKEN_REFRESH, 123
+            )
+
+        # Another session attempting a non-conflicting lock should succeed
+        try_advisory_transaction_lock(
+            other_db_session, LockType.OAUTH2_TOKEN_REFRESH, 456
+        )
+
+        # After the transaction ends, the lock should be released and other
+        # sessions should be able to acquire it.
+        db_session.rollback()
+        try_advisory_transaction_lock(
+            other_db_session, LockType.OAUTH2_TOKEN_REFRESH, 123
+        )
```

Comment:
I think this might be four separate tests for four different situations:

1. Just successfully getting a lock that no one else has
2. Failing to get a lock because someone else already has it
3. Successfully getting a lock while someone else has a non-conflicting lock
4. Successfully getting a lock after someone else has released it

Same principle again: "a test should have only one reason to fail"